### ISSUE TITLE
Add quasi-1D multicomponent Euler semidiscretization

### ISSUE THREAD

**Issue by stillyslalom** [stillyslalom/TrixiShockTube.jl#8]
Laboratory shock tubes often have lengthwise variations in cross-sectional area - see https://www.tfd.chalmers.se/~nian/courses/compflow/docs/Quasi-One-Dimensional-Flow_Governing-Equations.pdf

**Comment by stillyslalom** [comment]
@jlchan do you intend to upstream the quasi-1D semidiscretization described in https://www.sciencedirect.com/science/article/pii/S0021999124001256 to Trixi?

**Comment by jlchan** [comment]
@stillyslalom yes, it was actually implemented in https://github.com/trixi-framework/Trixi.jl/pull/1757 (with some bug fixes afterwards).

**Comment by stillyslalom** [comment]
Great, thanks! I'll look at extending it to multicomponent flows.

**Comment by jlchan** [comment]
Nice! Let me know if you have any questions, and good luck
